adserver/models.py

```diff
@@ -974,6 +1002,31 @@ def show_to_geo(self, geo_data):
             ):
                 return False
 
+        # Check if the country traffic cap exceeds the current fill for that country
+        if self.traffic_cap and "countries" in self.traffic_cap:
+            # pylint: disable=invalid-sequence-index
+            limited_countries = self.traffic_cap["countries"]
+            traffic_fill = self.traffic_fill or {}
+            country_traffic_fill = traffic_fill.get("countries", {})
+            if country_traffic_fill.get(geo_data.country, 0.0) > limited_countries.get(
+                geo_data.country, 100.0
+            ):
+                return False
+
+        # Check if the region traffic cap exceeds the current fill for that region
+        if self.traffic_cap and "regions" in self.traffic_cap:
+            # pylint: disable=invalid-sequence-index
+            limited_regions = self.traffic_cap["regions"]
+            traffic_fill = self.traffic_fill or {}
```

Comment:
Seems like it might be cleaner to just do `if self.traffic_cap and self.traffic_fill` in initial `if` block, then we can assume this value exists, and don't execute if it isn't set.